fe/fe-core/src/main/java/org/apache/doris/statistics/util/InternalSqlTemplate.java

```diff
@@ -0,0 +1,282 @@
+    public static final String PARTITION_MAX_SIZE_SQL = "SELECT MAX(LENGTH(${column})) AS max_size FROM"
+            + " ${table} PARTITION (${partition});";
+
+    public static final String AVG_SIZE_SQL = "SELECT AVG(LENGTH(${column})) AS avg_size FROM ${table};";
+    public static final String PARTITION_AVG_SIZE_SQL = "SELECT AVG(LENGTH(${column})) AS avg_size"
+            + " FROM ${table} PARTITION (${partition});";
+
+    public static final String MAX_AVG_SIZE_SQL = "SELECT MAX(LENGTH(${column})) AS max_size,"
+            + " AVG(LENGTH(${column})) AS avg_size FROM ${table};";
+    public static final String PARTITION_MAX_AVG_SIZE_SQL = "SELECT MAX(LENGTH(${column}))"
+            + " AS max_size, AVG(LENGTH(${column})) AS avg_size FROM ${table} PARTITION (${partition});";
+
+    public static final String NUM_NULLS_SQL = "SELECT COUNT(1) AS num_nulls FROM ${table}"
+            + " WHERE ${column} IS NULL;";
+    public static final String PARTITION_NUM_NULLS_SQL = "SELECT COUNT(1) AS num_nulls FROM"
+            + " ${table} PARTITION (${partition}) WHERE ${column} IS NULL;";
+    /** ---------------------------- for statistics end ---------------------------- */
+
+    private static final Logger LOG = LogManager.getLogger(InternalSqlTemplate.class);
+
+    private static final Pattern PATTERN = Pattern.compile("\\$\\{\\w+\\}");
+
+    /**
+     * Concatenate SQL statements based on templates and parameters. e.g.:
+     * template and parameters:
+     *  'SELECT ${col} FROM ${table} WHERE id = ${id};',
+     *   parameters: {col=colName, table=tableName, id=1}
+     *   result sql: 'SELECT colName FROM tableName WHERE id = 1;
+     * <p>
+     *
+     * @param template sql template
+     * @param params   k,v parameter, if without parameter, params should be null
+     * @return SQL statement with parameters concatenated
+     */
+    public static String processTemplate(String template, Map<String, String> params) {
+        Matcher matcher = PATTERN.matcher(template);
+        StringBuffer sb = new StringBuffer();
+
+        while (matcher.find()) {
+            String param = matcher.group();
+            String value = params.get(param.substring(2, param.length() - 1));
+            matcher.appendReplacement(sb, value == null ? "" : value);
+        }
+
+        matcher.appendTail(sb);
+        LOG.debug("Template:{}, params: {}, SQL: {}", template, params, sb.toString());
+
+        return sb.toString();
+    }
+
+    private static boolean checkParams(Set<String> requiredParams, Map<String, String> params) {
+        if (params != null) {
+            Set<String> paramsSet = params.keySet();
+            return paramsSet.containsAll(requiredParams);
+        } else {
+            return requiredParams == null;
+        }
+    }
+
+    public static String buildStatsMinValueSql(Map<String, String> params) throws InvalidFormatException {
+        Set<String> requiredParams = Sets.newHashSet("table", "column");
```

Comment:
We'd better to wrap the requiredParams and SQL template string to a individual class rather  than hard code it in every single method.

Reply:
sure.